src/scenes/Partnerships/PartnershipForm/renderFields.tsx

```diff
@@ -0,0 +1,118 @@
+import { FormApi } from 'final-form';
+import React, { useEffect, useRef } from 'react';
+import {
+  hasManagingType,
+  PartnershipFormFragment,
+  PartnershipformValues,
+} from '.';
+import {
+  displayFinancialReportingType,
+  displayPartnershipStatus,
+  PartnershipAgreementStatus,
+  PartnershipAgreementStatusList,
+  PartnerType,
+} from '../../../api';
+import {
+  EnumField,
+  EnumFieldProps,
+  SecuredField,
+  SubmitError,
+} from '../../../components/form';
+import { PartnerField } from '../../../components/form/Lookup';
+import { Nullable } from '../../../util';
+
+export const renderFields = (partnership?: PartnershipFormFragment) => ({
+  values,
+  form,
+}: {
+  values: PartnershipformValues;
+  form: FormApi;
+}) => {
+  const lookupPartnerObj = values.partnership?.partnerLookupItem;
+  const lookupPartnerTypes = lookupPartnerObj?.types.value;
+  const lookupPartnerFinType = lookupPartnerObj?.financialReportingTypes.value;
+
+  const currentPartnerTypes = partnership?.partner.value?.types.value;
+  const currentPartnerFinTypes =
+    partnership?.partner.value?.financialReportingTypes.value;
+
+  const prevPartnerId = useRef<string>();
+  const prevTypes = useRef<Nullable<readonly PartnerType[]>>();
+
+  useEffect(() => {
+    // if user selects a different partner (on create partnership), wipe the types and fin type values
+    if (lookupPartnerObj?.id !== prevPartnerId.current) {
+      form.change('partnership.types', undefined);
```

Comment:
I'm seeing this behavior here, which seems like not how this is intended to work?

![partnership](https://user-images.githubusercontent.com/728777/97344400-12205280-185f-11eb-8adb-7fd130effa16.gif)